### ISSUE TITLE
Teleport to spawn on join not working as expected

### ISSUE THREAD

**Issue by Themis3000** [rockquiet/Spawn#11]
Hi there! I'm having some troubles configuring this plugin to do what I expect. I'm running a 3 server network. All 3 servers running paper, the "hub" server is the one running this plugin. I'm using Velocity as the proxy for the network.

Here's a relevant snippet of the config file:

```yml
# teleport the player to spawn on join
teleport-on-join:
  enabled: true
  # teleport the player ONLY on first join
  only-first-join: false
```

What I expect to happen: Whenever the player joins the hub, the player is teleported to spawn.

What is actually happening: The player maintains their last position.

I've already restarted the server and such. The /spawn set and /spawn commands are working as intended.

Let me know if there's anything else I'm supposed to do to achieve my goal, or if that goal is something this plugin even aims to solve. Thank you for your time and work on this plugin!

**Comment by rockquiet** [comment]
Hi, sorry to hear that the plugin is not working as expected. Your goal is actually one of the use cases I created the plugin for.

I've just tested it on Paper 1.21.1 (build 132) and 1.21.3 (build 77) behind Velocity 3.4.0 (build 452) and it works as expected on my end.
Could you please try if this solves the problem:

If you are using the `world-list`, make sure it does not accidentally prevent this feature from working.

All features have [bypass permissions](https://github.com/rockquiet/Spawn/wiki/Commands-&-Permissions#join-teleport-bypass) (`spawn.bypass.join-teleport`, `spawn.bypass.world-list` are relevant in your case) which are currently given to op's by default. You can negate this permission by setting it to `false` with a permission plugin.

If this is not the case and even non op players are not teleported, please check if another plugin is forcing them to their last position. You can also test if this happens with most/all other plugins disabled.

Please let me know if any of these steps have solved the problem or if you need further assistance.

**Comment by Themis3000** [comment]
Thank you for your help & for testing it out for me! I got it working properly, it turned out to be a combination of two things. I had removed some of the options I knew I'd never use from the config file to keep it slim, which seemed to cause the plugin to behave improperly sometimes (I noticed it throwing tracebacks in the console eventually). The other part is I didn't realize that op privileges where overriding it. I was so focused on trying to get the basic function of the plugin working before thinking about permissions that I never looked at that section of the docs (sorry! I should have considered that).
